### PR TITLE
Use PVC name and namespace for changing PVC conditions

### PR DESCRIPTION
**Pull request by ELENAGER**
Previously we used VolumeReplication name and namespace for updating PVC conditions due to the fact, that we had one-to-one relationship between VolumeReplication and PVC. Now when we are going to use consistency group this handling will not be relevant anymore, because several PVCs can be protected by one VolumeReplication. So, we need to use PVC name and namespace for updating PVC conditions.

### CODE REVIEW

**Review by nirs — COMMENTED** [review]
Looking the code it is not clear what is the old way and the new way - what we are trying to do.

Can you share an example resource with both old and new format?

Also I see hat no tests are modified - we don't have any tests fro this code? This is a good time to add some tests. The tests will also make it clear what the code is trying to do.

**Comment by ELENAGER** [comment]
The previous format of changing PVC conditions during reconciliation:
v.updatePVCDataReadyCondition(**volRep.Namespace**, **volRep.Name**, VRGConditionReasonReady, msg)

the new format will be:
v.updatePVCDataReadyCondition(**pvc.Namespace**, **pvc.Name**, VRGConditionReasonReady, msg)

**Comment by BenamarMk** [comment]
> Looking the code it is not clear what is the old way and the new way - what we are trying to do.

> Can you share an example resource with both old and new format?

> Also I see hat no tests are modified - we don't have any tests fro this code? This is a good time to add some tests. The tests will also make it clear what the code is trying to do.

@nirs what @ELENAGER is updating in this PR is the method we use to search for `vrg.Status.ProtectedPVCs`. Currently, we find a `ProtectedPVC` using the VR's name and namespace, but in the future, that will no longer be a valid approach for finding `ProtectedPVCs` in the `vrg.Status.ProtectedPVCs` list. A VR's name/namespace may not always be the same as the PVC that is protecting.

File: internal/controller/vrg_volrep.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func (v *VRGInstance) reconcileVRAsSecondary(pvc *corev1.PersistentVolumeClaim,
 
 	pvcNamespacedName := types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}
 
-	requeueResult, ready, err := v.processVRAsSecondary(pvcNamespacedName, log)
+	requeueResult, ready, err := v.processVRAsSecondary(pvcNamespacedName, pvc, log)
```
Comment (BenamarMk) [inline internal/controller/vrg_volrep.go:161]: Doesn't this contradict the intent of the PR? We are passing the PVC name and namespace to create the VR, but we can't use `pvcNamespacedName` to locate the PVC. I understand your intent, but this approach will face the same issue you are trying to safeguard against.

Comment (ShyamsundarR) [reply internal/controller/vrg_volrep.go:161]: @BenamarMk I think today it is the same i.e VR name is the same as the PVC name and should remain so for backward compat. But, in the future the VR name would be a group name and the first argument would change to something other than the pvc name/ns.

The intent is to reuse `processVRAs*` to process a VR or a VRGroup, given a PVC.

Comment (BenamarMk) [reply internal/controller/vrg_volrep.go:161]: Agreed. Setting the groundwork in this PR is the first step.

**Review by BenamarMk — APPROVED** [review]

**Pull request merged**